src/main/java/com/google/ehub/servlets/EntertainmentItemServlet.java

```diff
@@ -0,0 +1,176 @@
+package com.google.ehub.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.common.collect.Sets;
+import com.google.ehub.data.EntertainmentItem;
+import com.google.ehub.utility.BlobstoreURLUtility;
+import com.google.ehub.utility.NextIncreasingLexicographicStringUtility;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/*
+ * Handles GET and POST requests to manage EntertainmentItem entities
+ * stored in Datastore.
+ */
+@WebServlet("/entertainment-item-data")
+public class EntertainmentItemServlet extends HttpServlet {
+  private static final String ENTERTAINMENT_ITEM_KIND = "entertainmentItem";
+  private static final String SORTING_MODE_KEY = "sortingMode";
+  private static final String SEARCH_VALUE_KEY = "searchValue";
+  private static final String TITLE_PROPERTY_KEY = "title";
+  private static final String LOWERCASE_TITLE_PROPERTY_KEY = "lowercaseTitle";
+  private static final String DESCRIPTION_PROPERTY_KEY = "description";
+  private static final String IMAGE_URL_PROPERTY_KEY = "imageURL";
+  private static final String TITLE_INCREASING_ALPHA_KEY = "TITLE_INCREASING_ALPHABETICAL_ORDER";
+  private static final String TITLE_DECREASING_ALPHA_KEY = "TITLE_DECREASING_ALPHABETICAL_ORDER";
+
+  private static final Set<String> SORTING_MODES =
+      Sets.newHashSet(TITLE_INCREASING_ALPHA_KEY, TITLE_DECREASING_ALPHA_KEY);
+
+  private static final int MAX_TITLE_CHARS = 64;
+  private static final int MAX_DESCRIPTION_CHARS = 2048;
+
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String searchValue = request.getParameter(SEARCH_VALUE_KEY);
+    String sortingMode = request.getParameter(SORTING_MODE_KEY);
+
+    if (!areGETRequestParametersValid(searchValue, sortingMode)) {
+      return;
```

Comment:
It might be helpful to add a print statement here that the parameters are invalid.

Reply:
I added print statements in last commmits.